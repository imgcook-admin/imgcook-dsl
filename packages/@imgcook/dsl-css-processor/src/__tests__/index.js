import { weex, web, Process as Processor } from '../';

test('exports web & weex as instance of Processor', () => {
  expect(web).toBeInstanceOf(Processor);
  expect(weex).toBeInstanceOf(Processor);
});

const testStyle = {
  fontSize: 20,
  marginTop: 10,
  marginLeft: '0',
  marginRight: '0px',
  marginBottom: 0,
  backgroundImage: '0px',
  color: '#FF5E28',
  letterSpacing: '0',
  lineHeight: '20rem',
  lines: 1,
  // fontWeight: 'normal',
  flexDirection: 'column',
  fontWeight: 400,
  fontFamily: 'PingFangSC',
};

test('web processor should function normally', () => {
  console.log(web.filter(testStyle));
  expect(web.filter(testStyle)).toMatchSnapshot();
});

test('weex processor should function normally', () => {
  expect(weex.filter(testStyle)).toMatchSnapshot();
});