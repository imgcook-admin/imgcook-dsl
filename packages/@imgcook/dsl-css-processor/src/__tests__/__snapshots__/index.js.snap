// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web processor should function normally 1`] = `
Object {
  "backgroundImage": "0px",
  "color": "#FF5E28",
  "flexDirection": "column",
  "fontSize": 20,
  "letterSpacing": "0",
  "lineHeight": "20rem",
  "marginTop": 10,
}
`;

exports[`weex processor should function normally 1`] = `
Object {
  "backgroundImage": "0px",
  "color": "#FF5E28",
  "flexDirection": "column",
  "fontSize": 20,
  "lineHeight": "20rem",
  "lines": 1,
  "marginTop": 10,
}
`;
